src/zeroband/inference/pipeline.py

```diff
@@ -0,0 +1,221 @@
+import time
+import pickle
+from typing import Tuple
+from functools import partial
+
+import torch
+import torch.nn as nn
+from vllm.model_executor.layers.sampler import SamplerOutput
+from pydantic_config import BaseConfig
+from vllm import LLM
+
+from prime_iroh import Node
+from zeroband.utils.logger import get_logger
+
+# Global logger
+logger = get_logger("INFER")
+
+
+class PipelineConfig(BaseConfig):
+    rank: int = 0
+    world_size: int = 1
+    iroh_seed: int | None = None
+    iroh_peer_id: str | None = None
```

Comment:
on alternative would be to do 
```suggestion
class PipelineConfig(BaseConfig):
    rank: int = 0
    world_size: int = 1
    iroh_seed: int 
    iroh_peer_id: str 
```

and do in the infer.py

```python
class Config(BaseConfig):
    pp: PipelineConfig | None = None
```

this would catch at config time the case where iroh_seed or iroh_peer_id are  None and world_size >1.

Altenativly we could check that if world_size > 1 iroh_seed and iroh_peer_id should not be None, but I feel that its hacky

Reply:
| this would catch at config time the case where iroh_seed or iroh_peer_id are None and world_size >1.

but this is actually expected behavior. if the seed and peer id are not specified the nodes generate a node id randomly, and wait via stdin to connect to correct neighbors

Reply:
oh I see, then sounds good